src/vs/workbench/contrib/terminal/browser/terminalEditorInput.ts

```diff
@@ -23,6 +25,15 @@ export class TerminalEditorInput extends EditorInput {
 		return TerminalEditor.ID;
 	}
 
+	override copy(): IEditorInput {
```

Comment:
@lramos15 @bpasero would it be possible to make `copy` return an optional `Promise`? This is needed to get the `terminal.integrated.splitCwd` setting working here since we only query the shell's current cwd when requested to avoid polling.

Reply:
Fine by me but probably ripples through a lot of code that is currently sync.